hera_cal/abscal.py

```diff
@@ -981,510 +2636,319 @@ def TT_Phi_gain_arr(self):
+                                     gain_convention=gain_convention, flags_are_wgts=True)
+        for k in gains_to_update.keys():
+            gains_to_update[k] *= gains_here[k]
+        if max_iter > 1:
+            crit = np.median(np.linalg.norm([gains_here[k] - 1.0 for 
+                                             k in gains_here.keys()], axis=(0, 1)))
+            echo("phase_slope_cal convergence criterion: " + str(crit), verbose=verbose)
+            if crit < phs_conv_crit:
+                break
+
+
+def post_redcal_abscal(model, data, flags, rc_flags, min_bl_cut=None, max_bl_cut=None, edge_cut=0, 
+                       tol=1.0, gain_convention='divide', phs_max_iter=100, phs_conv_crit=1e-6, verbose=True):
+    '''Performs Abscal for data that has already been redundantly calibrated.
+
+    Arguments:
+        model: DataContainer containing externally calibrated visibilities, LST-matched to the data
+        data: DataContainer containing redundantly but not absolutely calibrated visibilities
+        flags: DataContainer containing combined data and model flags
+        rc_flags: dictionary mapping keys like (1, 'Jxx') to flag waterfalls from redundant calibration
+        min_bl_cut : float, eliminate all visibilities with baseline separation lengths
+            smaller than min_bl_cut. This is assumed to be in ENU coordinates with units of meters.
+        max_bl_cut : float, eliminate all visibilities with baseline separation lengths
+            larger than max_bl_cut. This is assumed to be in ENU coordinates with units of meters.
+        edge_cut : integer number of channels to exclude at each band edge in delay and global phase solvers
+        tol: float distance for baseline match tolerance in units of baseline vectors (e.g. meters)
+        gain_convention: either 'divide' if raw data is calibrated by dividing it by the gains
+            otherwise, 'multiply'.
+        phs_max_iter: maximum number of iterations of phase_slope_cal or TT_phs_cal allowed
+        phs_conv_crit: convergence criterion for updates to iterative phase calibration that compares
+            the updates to all 1.0s. 
+
+    Returns:
+        abscal_delta_gains: gain dictionary mapping keys like (1, 'Jxx') to waterfalls containing 
+            the updates to the gains between redcal and abscal
+        AC: AbsCal object containing absolutely calibrated data, model, and other useful metadata
+    '''
+    abscal_delta_gains = {ant: np.ones_like(g, dtype=complex) for ant, g in rc_flags.items()}
+
+    # instantiate Abscal object
+    wgts = DataContainer({k: (~flags[k]).astype(np.float) for k in flags.keys()})
+    AC = AbsCal(model, data, wgts=wgts, antpos=data.antpos, freqs=data.freqs, 
+                refant=pick_reference_antenna(synthesize_ant_flags(flags))[0],
+                min_bl_cut=min_bl_cut, max_bl_cut=max_bl_cut)
+    AC.antpos = data.antpos
+
+    # Global Delay Slope Calibration
+    for time_avg in [True, False]:
```

Comment:
haven't exposed `time_avg` kwarg, and why do you need the conditional?

Reply:
I thought the idea was to get an approximate global delay slope (assuming it doesn't change) by doing the average, then do it again for each integration. That's exactly what the old abscal run did.